portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -59,13 +61,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
     String userComment = request.getParameter("user-comment");
     long timestamp = System.currentTimeMillis();
-    if (userComment.isEmpty()) return;
+    String userEmail = userService.getCurrentUser().getEmail();
+    if (userComment.replaceAll("\\s","").isEmpty()) return;
```

Comment:
Won't this replace any whitespace including in between words?
Maybe you want to use trim() instead?

Reply:
Yes, however since this is just checking whether the submitted comment is empty, it should not make a difference. However, i can make it so that it trims first before posting as well, since I do not do any processing on the comment

Reply:
Ah I think I was confused due to the replaceAll.
I think the best alternative is to use .contains()?